init-tools.sh

```diff
@@ -14,4 +14,35 @@ while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symli
 done
 DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 
-$DIR/scripts/obtain/dotnet-install.sh --channel $CHANNEL --verbose
\ No newline at end of file
+$DIR/scripts/obtain/dotnet-install.sh --channel $CHANNEL --verbose
+
+__init_tools_log=$DIR/init-tools.log
+__DOTNET_CMD=$DOTNET_INSTALL_DIR/dotnet
+__BUILD_TOOLS_DIR=$DIR/build_tools
+__BUILD_TOOLS_SOURCE=https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json
+__BUILD_TOOLS_PACKAGE_VERSION=$(cat $DIR/BuildToolsVersion.txt)
+__BUILD_TOOLS_PATH=$NUGET_PACKAGES/Microsoft.DotNet.BuildTools/$__BUILD_TOOLS_PACKAGE_VERSION/lib
+__BUILD_TOOLS_SEMAPHORE=$__BUILD_TOOLS_DIR/init-tools.completed
```

Comment:
This file doesn't ever appear to be checked.  It is just created, but no where in the script below checks for its existence.